src/oc-install.ts

```diff
@@ -227,7 +227,13 @@ export class InstallHandler {
 
     tl.debug(`expanding ${archivePath} into ${downloadDir}`);
 
-    const archiveType = path.extname(archive);
+    let archiveType = path.extname(archive);
+    const expandDir = archive.replace(archiveType, '');
+    // handle tar.gz explicitly
+    if (path.extname(expandDir) === '.tar') {
+      archiveType = '.tar.gz';
```

Comment:
If the value of `archive` is `archiv.tar.bz2`, archiveType would not be correctly set.

I would have suggested to use `path.basename` instead of replace:
``` javascript
let archiveType = path.extname(archive);
if (path.extname(path.basename(archive)) === '.tar') {
  archiveType = '.tar' + archiveType;
}
```

Reply:
good point. i'll open a new issue

Reply:
And i think a `getArchiveType` method would be appropriate :)